components/admin_console/license_settings/license_settings.test.tsx

```diff
@@ -112,24 +113,26 @@ describe('components/admin_console/license_settings/LicenseSettings', () => {
             upgradeToE0Status: jest.fn().mockImplementation(() => Promise.resolve({percentage: 0, error: null})),
         };
         const props = {...defaultProps, enterpriseReady: false, actions};
-        const wrapper = shallow(<LicenseSettings {...props}/>);
+        const wrapper = shallow<LicenseSettings>(<LicenseSettings {...props}/>);
 
         expect(actions.getLicenseConfig).toBeCalledTimes(1);
         expect(actions.upgradeToE0Status).toBeCalledTimes(1);
         actions.upgradeToE0Status = jest.fn().mockImplementation(() => Promise.resolve({percentage: 1, error: null}));
 
         const instance = wrapper.instance();
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
         expect(instance.interval).toBe(null);
         expect(wrapper.state('upgradingPercentage')).toBe(0);
 
-        await instance.handleUpgrade({preventDefault: jest.fn()});
+        await instance.handleUpgrade({preventDefault: jest.fn()} as unknown as React.MouseEvent<HTMLButtonElement>);
         expect(actions.upgradeToE0).toBeCalledTimes(1);
         expect(actions.upgradeToE0Status).toBeCalledTimes(1);
         wrapper.update();
-        jest.setTimeout(() => {
-            expect(wrapper.update().state('upgradingPercentage')).toBe(1);
-            expect(instance.interval).not.toBe(null);
-        }, 100);
+        expect(wrapper.update().state('upgradingPercentage')).toBe(1);
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
```

Comment:
`interval` is private. I didn't want to change the component so  I put comment to ignore typescript, please let me know if there is any other way to handle this.